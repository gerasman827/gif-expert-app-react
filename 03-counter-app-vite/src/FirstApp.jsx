/* export function App() {
    return <h1>Hola mundo</h1>
}
 */
/* import React from 'react' */

// este no viene por defecto en vite pero si en reactApp
//sirve para definir el tipo a las props
import PropTypes from 'prop-types';

/* const result = (nombre) => {
  return "Me llamo "+nombre;
}
 */

export const FirstApp = ( { title, subTitle, name }) => {
    
  return (
    <>
      <h1> {title} </h1>
      {/* <div data-testid="test-title"> {title} </div> */}
      <h2> {name} </h2>   
      <p>{subTitle} </p>
      <p>{subTitle} </p>
    </>
  );
};

//para definir los tipos de los propTypes
FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string,

}

FirstApp.defaultProps = {
  title: "Germán Arenas Sánchez",
  subTitle:  'SOy un subtítulo',
  title: 'No hay título',
}