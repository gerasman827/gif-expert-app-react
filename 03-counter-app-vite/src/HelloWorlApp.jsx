export const HelloWorldApp = () => {
    return (
        <h2>Hello world: GermÃ¡n</h2>
    )
}