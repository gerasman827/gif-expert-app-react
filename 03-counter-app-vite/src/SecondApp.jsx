export const SecondApp = () => {
    return (
        <h2>Este es mi segundo componente en React</h2>
    )
}