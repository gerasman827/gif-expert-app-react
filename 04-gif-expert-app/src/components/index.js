//archivo de barril. Para manejar las exportaciones. Propio de JS

export * from './AddCategory'
export * from './GifGrid';
export * from './GifItem'