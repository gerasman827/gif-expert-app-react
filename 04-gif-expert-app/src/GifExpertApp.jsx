import { useState } from 'react';
import { AddCategory, GifGrid } from './components'

export const GifExpertApp = () => {
  const [categories, setCategories] = useState(['One Punch']);
  const onAddCategory = (newCategory) => {
    //el push en .js realiza mutaciones del objeto y en react se debe evitar hacer mutaciones
    /* categories.push("Valorant"); */

    //una forma de hacerlo
    /* setCategories([...categories, 'Valorant']); */

    //ahora, validar que la categoria nueva no exista en el arreglo
    //una forma:
    if (categories.includes(newCategory)) return;

    //otra forma
    setCategories((categories) => [newCategory, ...categories]);
  };

  return (
    <>
      {/* Titulo */}
      <h1>GifExpertApp</h1>

      {/* Input */}
      <AddCategory
        //esto oculta la implementaciÃ³n
        //setCategories={ setCategories }
        onNewCategory={(value) => onAddCategory(value)}
      />

      {
        categories.map((category) =>(
            <GifGrid
                key={ category }
                category={ category }
            />
        ))
      }
    </>
  )
}
