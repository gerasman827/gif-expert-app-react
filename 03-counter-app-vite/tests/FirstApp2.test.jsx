import { render, screen } from "@testing-library/react"
import { FirstApp } from "../src/FirstApp"

const title = 'Hola, soy Goku';
const subTitle = "Soy un subtítulo";

describe('Pruebas en FirstApp', () => { 
  test('Debe de hacer match con el snapsho', () => { 
    
    const {container} = render(<FirstApp title={title}/>);
    expect( container ).toMatchSnapshot();
   });

   test('Debe de mostrar el mensaje "Hola, soy Goku"', () => { 
      render( <FirstApp title={title}/>);
      /* screen.debug(); */
      expect( screen.getByText(title) ).toBeTruthy();
    });

    test('DEbe mostrar el título en un h1', () => { 
      render( <FirstApp title={title}/>);
      expect( screen.getByRole('heading',{level:1} ).innerHTML).toContain(title);
     });

     test('Debe de mostrar el subtítulo enviado por props', () => { 
        render(
        <FirstApp
          title={title}
          subTitle={subTitle}
        />);

        expect( screen.getAllByText(subTitle).length).toBe(2);
      })
 });