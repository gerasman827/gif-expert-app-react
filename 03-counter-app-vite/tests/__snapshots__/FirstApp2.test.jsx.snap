// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp Debe de hacer match con el snapsho 1`] = `
<div>
  <h1>
     
    Hola, soy Goku
     
  </h1>
  <h2>
     
     
  </h2>
  <p>
    SOy un subtítulo
     
  </p>
  <p>
    SOy un subtítulo
     
  </p>
</div>
`;
